logger.go

```diff
@@ -4,22 +4,130 @@ import (
 	"os"
 	"time"
 
+	"github.com/blendle/zapdriver"
+	"github.com/prometheus/client_golang/prometheus"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
-func newLogger(level zapcore.Level, encoder zapcore.Encoder) (*zap.Logger, zap.AtomicLevel) {
+func (s *SVC) newLogger(level zapcore.Level, encoder zapcore.Encoder) (*zap.Logger, zap.AtomicLevel) {
 	atom := zap.NewAtomicLevel()
 	atom.SetLevel(level)
 
+	zapOpts := append(s.zapOpts, zap.ErrorOutput(zapcore.Lock(os.Stderr)), zap.AddCaller())
+
 	logger := zap.New(zapcore.NewSampler(zapcore.NewCore(
 		encoder,
 		zapcore.Lock(os.Stdout),
 		atom,
 	), time.Second, 100, 10),
-		zap.ErrorOutput(zapcore.Lock(os.Stderr)),
-		zap.AddCaller(),
+		zapOpts...,
 	)
 
 	return logger, atom
 }
+
+func WithZapMetrics() Option {
+	return func(s *SVC) error {
+		requestCount := prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "logger_emitted_entries",
+				Help: "Number of log messaged emitted.",
```

Comment:
So close. Typo: `messages`

Reply:
And we even have the misspell linter enabled. :thinking: